python/pyarrow/tests/test_array.py

```diff
@@ -1057,6 +1057,31 @@ def test_map_from_arrays():
 
     assert result.equals(expected)
 
+    result = pa.MapArray.from_arrays(offsets, keys, items, pa.map_(
+        keys.type,
+        items.type
+    ))
+    assert result.equals(expected)
+
+    pa.MapArray.from_arrays(offsets, keys, items, pa.map_(
+        keys.type,
+        # Bigger than the original i4, which is okay
+        pa.int32()
+    ))
+    assert result.equals(expected)
```

Comment:
Does this assertion serve a purpose?

Reply:
Yeah, I think `result = ...` is missing in line 1066.

And for the type-double-check, `pa.int32` and numpy `i4` should be same?
If we were to use `pa.int64` the construction of a MapArray should not fail, but the assertion would, right?

Reply:
Thanks! I'm not too familiar with the `numpy` types, thanks for pointing that out!